pkg/controller/extendeddaemonsetreplicaset/strategy/rollingupdate.go

```diff
@@ -53,7 +53,7 @@ func ManageDeployment(client client.Client, params *Parameters) (*Result, error)
 				nbIgnoredUnresponsiveNodes++
 				continue
 			}
-			if !compareSpecTemplateMD5Hash(params.Replicaset.Spec.TemplateGeneration, pod) {
+			if !compareCurrentPodWithNewPod(params.EDSName, params.Replicaset, pod, node) {
```

Comment:
should we make  `compareCurrentPodWithNewPod` accept `Parameters` as an argument here instead of `params.EDSName` and `params.Replicaset` ? 
```suggestion
			if !compareCurrentPodWithNewPod(params, pod, node) {
```